probe_scraper/probe_expiry_alert.py

```diff
@@ -109,6 +110,12 @@ def send_emails_for_expiring_probes(expired_probes, expiring_probes,
     logging.info(f"Sent emails to {len(probes_by_email_by_state)} recipients")
 
 
+def download_file(url, output_filename):
+    content = requests.get(url).text
+    with open(output_filename, "w") as scalars_file:
+        scalars_file.write(content)
```

Comment:
I feel like this should use a temp file, e.g. use [NamedTemporaryFile](https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile) and accept a parse method:
```python
from functools import partial
from tempfile import NamedTemporaryFile

...

def download_file(url, parse):
    content = requests.get(url).text
    with NamedTemporaryFile(text=True) as temp_file:
        temp_file.write(content)
        return parse([temp_file.name])

...

def main(current_date, dryrun):
    current_version = get_latest_nightly_version()
    next_version = str(int(current_version) + 1)

    events = download_file(BASE_URI + EVENTS_FILE, EventsParser().parse)

    histograms = download_file(
        BASE_URI + HISTOGRAMS_FILE,
        partial(HistogramsParser().parse, version=next_version),
    )

    scalars = download_file(BASE_URI + SCALARS_FILE, ScalarsParser().parse)
```

Reply:
The histogram parser only accepts certain filenames.  I do agree that something temporary should be used though.  Instead I used `gettempdir()`